pkg/operator/types/types.go

```diff
@@ -0,0 +1,138 @@
+package types
+
+import (
+	"context"
+	"io"
+
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	walrus "github.com/seal-io/walrus/pkg/apis/walrus/v1"
+	walruscore "github.com/seal-io/walrus/pkg/apis/walruscore/v1"
+	"github.com/seal-io/walrus/pkg/clients/clientset"
+)
+
+// Type indicates the type of Operator,
+// e.g. Kubernetes, AWS, etc.
+type Type = string
+
+// Operator holds the actions that an operator must satisfy.
+type Operator interface {
```

Comment:
The name can cause confusion in the context of Kubernetes. Let's change it to ResourceHandler.

And adapt with existing packaging pattern. Move the interface to pkg resourcehandler and implementations to resourcehandlers.